modules/codegen/src/smithy4s/codegen/transformers/AwsConstraintsRemover.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.codegen.transformers
+
+import software.amazon.smithy.build.ProjectionTransformer
+import software.amazon.smithy.build.TransformContext
+import software.amazon.smithy.model.Model
+
+/** A preprocessor that removes constraints from specifications of AWS. */
+class AwsConstraintsRemover extends ProjectionTransformer {
+
+  override def getName: String = AwsConstraintsRemover.name
+
+  private val constraintsToRemove = List("length", "range", "pattern")
```

Comment:
I just picked these 3 out initially from the list of constraint traits, but if any more need to be added, let me know. So far I've only had problems with `length` and `range`

Reply:
Any reason to avoid including all listed here https://smithy.io/2.0/spec/constraint-traits.html

Reply:
side note, no clue why `private` is considered a constraint trait because it doesn't conform to the definition:

> Constraint traits are used to constrain the values that can be provided for a shape.

it's a smithy model thing only.

Reply:
> Any reason to avoid including all listed here https://smithy.io/2.0/spec/constraint-traits.html

I had some doubts if the other ones really needed to be included, so I put it up like this and if needed I'll just add it more! :P

Reply:
Others don't need to be included : the only reason we're including those is because of how they are treated in smithy4s.